src/components/todo-list-element-builder.components.ts

```diff
@@ -0,0 +1,270 @@
+/**
+ * @description UI 생성 클래스
+ */
+import { DEFAULT_LABEL, EVENT_BUS_TYPES } from '../constants/todo-list.const';
+import {
+  Utils,
+  type EventBusType,
+  type EventsPayload,
+  type TodoListAppOptions,
+  type TodoListItem,
+} from '../types/todo.types';
+import { TodoListAppStyles } from './todo-list-styles.components';
+
+export class TodoListElementBuilder {
+  private options: TodoListAppOptions;
+  private initialLabel = DEFAULT_LABEL;
+  todoStyles: TodoListAppStyles;
+
+  dispatch: (event: EventBusType) => void;
+
+  constructor(
+    options: TodoListAppOptions,
+    styles: TodoListAppStyles,
+    eventBus: (e: EventBusType) => void,
+  ) {
+    this.options = options;
+    this.todoStyles = styles;
+    this.dispatch = eventBus;
+    if (this.options.labels) {
+      this.initialLabel = {
+        ...DEFAULT_LABEL,
+        ...this.options.labels,
+      };
+    }
+  }
+
+  createUl(events: EventsPayload[] = []) {
+    const ul = document.createElement('ul');
+    this.attachEvents(ul, events);
+    return ul;
+  }
+
+  createInput(events: EventsPayload[] = []) {
+    const input = document.createElement('input');
+    input.type = 'text';
+    this.attachEvents(input, events);
+    return input;
+  }
+
+  createRow(item: TodoListItem) {
+    const { id, label, isChecked } = item;
+    const dataId = id.toString();
+    const li = document.createElement('li');
+    li.className = `${this.todoStyles.clsNames.li} ${isChecked ? 'checked' : ''}`;
+
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    checkbox.checked = isChecked;
+    checkbox.id = dataId;
+
+    const lb = document.createElement('label');
+    lb.className = this.todoStyles.clsNames.label;
+    lb.textContent = label;
+    lb.htmlFor = checkbox.id;
+
+    [li, lb, checkbox].forEach((el) => el.setAttribute('data-id', dataId));
+    li.appendChild(checkbox);
+    li.appendChild(lb);
+    return li;
+  }
+
+  createNoItems() {
```

Comment:
```suggestion
  createEmptyRow() {
```
createEmptyRow가 어떨까요?